### PR TITLE
Integrates Auth with Predictions, Updates Sign Out, Disables AWS Logs

### PR DESCRIPTION
**Pull request by TrekSoft**
- Integrates Auth with Predictions
- Updates SignOut to do local by default with an option to explicitly sign out globally which will fail if the device is offline and fall back to local if a global sign out was first done on another device.
- Disables AWS logs to remove some of the confusing clutter in LogCat (AWSMobileClient still does it's own logs which is very annoying).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java [hunk 1]
```diff
@@ -119,6 +123,7 @@ public void configure(
     ) throws AuthException {
         final CountDownLatch latch = new CountDownLatch(1);
         final AtomicReference<Exception> asyncException = new AtomicReference<>();
+        LogFactory.setLevel(LogFactory.Level.OFF);
```
Comment (jpignata) [inline aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java:126]: I do wonder idly if we could do a kludge a thing where we listen to the logs and reply them all as `DEBUG`.  A bad idea for later maybe.

File: aws-predictions/src/main/java/com/amplifyframework/predictions/aws/AWSPredictionsPlugin.java [hunk 2]
```diff
@@ -79,7 +91,26 @@ public String getPluginKey() {
...
+        } else {
+            try {
+                credentialsProvider =
+                        (AWSMobileClient) Amplify.Auth.getPlugin(AUTH_DEPENDENCY_PLUGIN_KEY).getEscapeHatch();
+            } catch (IllegalStateException exception) {
+                throw new PredictionsException(
+                        "AWSPredictionsPlugin depends on AWSCognitoAuthPlugin but it is currently missing",
+                        exception,
+                        "Before configuring Amplify, be sure to add AWSPredictionsPlugin same as you added " +
+                                "AWSPinpointAnalyticsPlugin."
```
Comment (jpignata) [inline aws-predictions/src/main/java/com/amplifyframework/predictions/aws/AWSPredictionsPlugin.java:108]: Copypasta.

File: core/src/main/java/com/amplifyframework/auth/options/AuthSignOutOptions.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+ */
+public class AuthSignOutOptions {
+    private final boolean globalSignOut;
+
+    /**
+     * Advanced options for signing up.
+     * @param globalSignOut Sign out of all devices
+     */
+    protected AuthSignOutOptions(
+            boolean globalSignOut
```
Comment (jpignata) [inline core/src/main/java/com/amplifyframework/auth/options/AuthSignOutOptions.java:32]: This formatting seems curious?

File: aws-predictions/src/main/java/com/amplifyframework/predictions/aws/AWSPredictionsPlugin.java (hunk 2 above)
Comment (TrekSoft) [reply aws-predictions/src/main/java/com/amplifyframework/predictions/aws/AWSPredictionsPlugin.java:108]: Yeah gonna clean this all up after GA - it's all throughout the framework.

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java (hunk 1 above)
Comment (TrekSoft) [reply aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java:126]: Maybe - I kinda want to get away from logging stuff in favor of Hub and Callbacks but we should discuss as a team what we want our policy to be on that,

File: core/src/main/java/com/amplifyframework/auth/options/AuthSignOutOptions.java (hunk 3 above)
Comment (TrekSoft) [reply core/src/main/java/com/amplifyframework/auth/options/AuthSignOutOptions.java:32]: How so?

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java (hunk 1 above)
Comment (raphkim) [inline aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java:126]: Is this for turning off mobile client logs?

**Comment by raphkim** [comment]
Ah oops, turns out this is already reviewed haha

**Pull request merged**